ui/components/DataTable.tsx

```diff
@@ -161,31 +161,28 @@ function filterText(
 
   return _.filter(rows, (row) => {
     let matches = false;
-    for (const colName in row) {
-      const value = row[colName];
 
-      const field = _.find(fields, (f) => {
-        if (typeof f.value === "string") {
-          return f.value === colName;
-        }
-
-        if (f.sortValue) {
-          return f.sortValue(row) === value;
-        }
-      });
+    fields.forEach((field) => {
+      if (!field.textSearchable) return matches;
 
-      if (!field || !field.textSearchable) {
-        continue;
+      let value;
+      if (field.sortValue) {
+        value = field.sortValue(row);
+      } else {
+        typeof field.value === "function"
+          ? (value = field.value(row))
+          : (value = row[field.value]);
       }
 
-      // This allows us to look for a fragment in the string.
-      // For example, if the user searches for "pod", the "podinfo" kustomization should be returned.
-      for (const filterString of textFilters) {
-        if (_.includes(value, filterString)) {
+      for (let i = 0; i < textFilters.length; i++) {
+        if (value.includes(textFilters[i])) {
```

Comment:
This can be refactored as

```
matches = value.includes(textFilters[i]);

        if (!matches) {
          break;
        }
```

It would be easier to read.